nix/tools/lein/generate-nix.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+# This script takes care of generating/updating the nix files in this directory.
+# For this, we start with a clean cache (in ./.m2~/repository/) and call cljsbuild
+#  to cause it to download all the artifacts. At the same time, we note them
+#  in lein-project-deps-maven-inputs.txt so that we can use that as an input
+#  to maven-inputs2nix.sh
```

Comment:
I REALLY think one thing we need to do is extract the maven stuff into it's own repo, because this is really hard to follow. It needs to have clearly defined inputs and outputs, because I can't fully follow this.